app/models/vm_or_template.rb

```diff
@@ -231,6 +231,13 @@ def from_infra_manager?
   alias_method :parent_cluster, :ems_cluster
   alias_method :owning_cluster, :ems_cluster
 
+  def self.concrete_subclasses
+    leaf_subclasses | descendants.select { |d| d.try(:acts_as_sti_leaf_class?) }
+  end
+
+  # when looking at supported features, only look at the classes permitted to be used
+  singleton_class.send(:alias_method, :supported_subclasses, :concrete_subclasses)
+
```

Comment:
Thinking this should probably go in a mixin or `ApplicationRecord` since it is common for all classes which could use `ActsAsStiLeafClass`

Reply:
+1 for mixin, but I'm curious what it would look like. Definitely for a separate PR.